plugins/org.eclipse.fordiac.ide.gitlab/src/org/eclipse/fordiac/ide/gitlab/management/GitLabDownloader.java

```diff
@@ -94,7 +95,15 @@ public Map<Project, List<Package>> getProjectsAndPackages() {
 		return projectAndPackageMap;
 	}
 
-	public void fetchProjectsAndPackages() {
+	public DownloadResult<Void> fetchProjectsAndPackages() {
+		final boolean missingUrl = baseUrl == null || baseUrl.isBlank();
+		final boolean missingToken = token == null || token.isBlank();
+		if (missingUrl && missingToken) {
+			return new DownloadResult<>(DownloadResult.Status.NO_CONFIG, "No Configuration"); //$NON-NLS-1$
```

Comment:
As the string `"No Configuration"` is a message to the user it should be translateable - marking it als `NON-NLS` is not optimal

This applies for the message-attributes of the DownloadResult constructor in this pull request

@oberlehner 
 - what is the preferred way of dealing with strings that should be translated? directly create the Messages-class or leave the unmarked strings?
 - what is the preferred way of building complex messages from parts? Is MessageFormat still the way to go?

Reply:
Currently I would say we should not create any unmarked strings anymore. The externalize string wizard got really good and it is very quickly done. 

For most strings MessageFormat is the way to go. This allows to have both the base message and if needed the inserted strings translateable. The reason for using something like MessageFormat is that it allows that translations can reorder the insertion places so that it fits the grammar of the target language.

Reply:
Externalized the message strings